integration-tests/src/test/java/org/wso2/appserver/test/integration/TestSuiteListener.java

```diff
@@ -0,0 +1,181 @@
+
+    @Override
+    public void onStart(ISuite iSuite) {
+
+        try {
+            log.info("Starting pre-integration test setup...");
+
+            appserverHome = new File(System.getProperty(TestConstants.APPSERVER_HOME));
+            log.info("Application server home : " + appserverHome.toString());
+
+            serverStartCheckTimeout = Integer.valueOf(System.getProperty(TestConstants.SERVER_TIMEOUT));
+
+
+            if (isPortAvailable(TestConstants.TOMCAT_DEFAULT_PORT)) {
+                log.info("Default port " + TestConstants.TOMCAT_DEFAULT_PORT + " is available.");
+                System.setProperty(TestConstants.APPSERVER_PORT, String.valueOf(TestConstants.TOMCAT_DEFAULT_PORT));
+                applicationServerPort = Integer.valueOf(System.getProperty(TestConstants.APPSERVER_PORT));
+            }
+
+
+            log.info("Starting the server...");
+            applicationServerProcess = startPlatformDependApplicationServer();
+
+            waitForServerStartup();
+
+            if (isSuccessServerStartup) {
+                log.info("Application server started successfully. Running test suite...");
+            }
+
+        } catch (IOException ex) {
+            terminateApplicationServer();
+            String message = "Could not start the server";
+            log.error(message, ex);
+            throw new RuntimeException(message, ex);
+        }
+
+
+    }
+
+    @Override
+    public void onFinish(ISuite iSuite) {
+
+        log.info("Starting post-integration tasks...");
+        log.info("Terminating the Application server");
+        terminateApplicationServer();
+        log.info("Finished the post-integration tasks...");
+    }
+
+    private Process startPlatformDependApplicationServer() throws IOException {
+        String os = System.getProperty("os.name");
+        log.info(os + " operating system was detected");
+        if (os.toLowerCase().contains("unix") || os.toLowerCase().contains("linux")) {
+            log.info("Starting server as a " + os + " process");
+            return applicationServerProcess = new ProcessBuilder()
+                    .directory(appserverHome)
+                    .command("./bin/catalina.sh", "run")
+                    .start();
+        } else if (os.toLowerCase().contains("windows")) {
+            log.info("Starting server as a " + os + " process");
+            return applicationServerProcess = new ProcessBuilder()
+                    .directory(appserverHome)
+                    .command("\\bin\\catalina.bat", "run")
+                    .start();
+        }
+        return null;
+    }
+
+    public void terminateApplicationServer() {
+        if (applicationServerProcess != null) {
+            applicationServerProcess.destroy();
+        }
+    }
+
+    private void waitForServerStartup() throws IOException {
+
+        log.info("Checking server availability... (Timeout: " + serverStartCheckTimeout + " seconds)");
+        int currentSeconds = 0;
```

Comment:
I believe we need to rename the variable here.. What does currentSeconds refer to? As I understand, this is to set a ceiling on maximum iterations you will wait for the port to be available.